### PR TITLE
CRM-20424 Add method to mark deprecation & ensure it causes test fail

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
* [CRM-20424: Set up method for marking code as deprecated](https://issues.civicrm.org/jira/browse/CRM-20424)

### CODE REVIEW

**Comment by seamuslee001** [comment]
Jenkins test this please

**Comment by eileenmcnaughton** [comment]
@totten FYI - this is probably at the simplest end of what we discussed but I think we should merge as is & consider more complexity later if we want

File: CRM/Core/OptionGroup.php [hunk 1]
```diff
@@ -342,6 +342,7 @@ public static function lookupValues(&$params, $names, $flip = FALSE) {
    * @return null
    */
   public static function getLabel($groupName, $value, $onlyActiveValue = TRUE) {
+    Civi::log()->warning('Deprecated function, use CRM_Core_Pseudoconsant::getLabel', array('civi.tag' => 'deprecated'));
```
Comment (totten) [inline CRM/Core/OptionGroup.php]: Type: `Pseudo[Cc]onstant`

**Comment by totten** [comment]
General +1 on this technique for emitting deprecation notices.

**Pull request merged**